README.md

```diff
@@ -154,6 +154,30 @@ const isCompany = isOfShape({
 isCompany({name: 'Untitled', users: [{name: 'John', age: 21}]) // => true
 ```
 
+#### Optional Properties
+
+If a property can optionally be defined, wrap the type guard for it in `optional`.
+
+```ts
+type User = { name: string; perferredName?: string };
+const isUser = isOfShape({ name: isString, perferredName: optional(isString) })
+
+isUser({name: 'John'}) // => true
+isUser({name: 'John', perferredName: 'Johnny'}) // => true
+isUser({name: 'John', perferredName: undefined}) // => true
```

Comment:
I'm don't think we should allow this to pass. This is _not_ the `User` type defined above (with `preferredName?: string`), but a `preferredName?: string | undefined`. It's better to use the `key in object` for dealing with optional properties on objects.

To get the behavior in this example, the consumer would do `preferredName: optional(isOneOf(isString, isUndefined))`, which is consistent with three “type hints” you write in typescript (`optional` is `?`, `isString` is `string`, `isUndefined` is `undefined`).

(Also there's a typo in `perferredName`, should be `preferredName` :smile:)

Reply:
I've updated to change that behavior and fix that typo.